controllers/cloud.redhat.com/pool_controller.go

```diff
@@ -149,25 +152,38 @@ func (r *NamespacePoolReconciler) getPoolStatus(ctx context.Context, pool crd.Na
 					errNamespaceList = append(errNamespaceList, ns.Name)
 				}
 			}
+
+			if val := ns.Annotations["env-status"]; val == "ready" || val == "creating" {
+				totalNS++
+			}
 		}
 	}
 
 	statusTypeCount["ready"] = readyNS
 	statusTypeCount["creating"] = creatingNS
+	statusTypeCount["total"] = totalNS
 
 	return statusTypeCount, errNamespaceList, nil
 }
 
-func (r *NamespacePoolReconciler) checkReadyNamespaceQuantity(pool crd.NamespacePool) int {
+func (r *NamespacePoolReconciler) checkNamespaceQuantity(pool crd.NamespacePool, statusTypeCount map[string]int) int {
 	size := pool.Spec.Size
+	limiter := pool.Spec.SizeLimiter
 	ready := pool.Status.Ready
 	creating := pool.Status.Creating
 
+	if limiter > 0 {
+		if statusTypeCount["total"] >= limiter {
+			r.Log.Info(fmt.Sprintf("Max number of namespaces for pool [%s] already created or in-use. Max of [%d] allowed.", pool.Name, limiter))
```

Comment:
So the problem here is that it only takes into account the number that are in the pool and the number being created, which actually should be equal to `size`. (The only time it's not if is the pool has a size `x` and the pool size is reduced by `y`, but that requires someone to make changes to the spec)

Actually what we really want here is a limit on the TOTAL number of namespaces that have come from that pool. So we should look at namespaces which are also checked out :)